api/apiclient.go

```diff
@@ -591,14 +592,27 @@ func dialAPI(info *Info, opts0 DialOpts) (*dialResult, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	dialInfo, err := dialWebsocketMulti(ctx, info.Addrs, path, opts)
+	dialInfo, err := dialWebsocketMulti(ctx, shuffleAddresses(info.Addrs), path, opts)
```

Comment:
I'm -1 on applying this logic to client (non-agent) connections. Perhaps move this logic to worker/apicaller?